lib/cable_ready/channel.rb

```diff
@@ -2,209 +2,25 @@
-    #   }, ...],
-    #
-    #   outer_html: [{
-    #     selector:      "string",
-    #     focus_selector: "string",
-    #     html:          "string"
-    #   }, ...],
-    #
-    #   text_content: [{
-    #     selector: "string",
-    #     text:     "string"
-    #   }, ...]
-    #
-    #   insert_adjacent_html: [{
-    #     selector:      "string",
-    #     focus_selector: "string",
-    #     position:      "string",
-    #     html:          "string"
-    #   }, ...],
-    #
-    #   insert_adjacent_text: [{
-    #     selector: "string",
-    #     position: "string",
-    #     text:     "string"
-    #   }, ...],
-    #
-    #   remove: [{
-    #     selector:      "string",
-    #     focus_selector: "string,
-    #   }, ...],
-    #
-    #   set_property: [{
-    #     name:     "string",
-    #     selector: "string",
-    #     value:    "string"
-    #   }, ...],
-    #
-    #   set_value: [{
-    #     selector: "string",
-    #     value:    "string"
-    #   }, ...],
-    #
-    #   # Attribute Mutations .........................................................................................
-    #
-    #   set_attribute: [{
-    #     selector: "string",
-    #     name:     "string",
-    #     value:    "string"
-    #   }, ...],
-    #
-    #   remove_attribute: [{
-    #     selector: "string",
-    #     name:     "string"
-    #   }, ...],
-    #
-    #   # CSS Class Mutations .........................................................................................
-    #
-    #   add_css_class: [{
-    #     selector: "string",
-    #     name:     "string"
-    #   }, ...],
-    #
-    #   remove_css_class: [{
-    #     selector: "string",
-    #     name:     "string"
-    #   }, ...],
-    #
-    #   # Style Mutations ...........................................................................................
-    #
-    #   set_style: [{
-    #     selector: "string",
-    #     name:     "string",
-    #     value:    "string"
-    #   }, ...],
-    #
-    #   # Dataset Mutations ...........................................................................................
-    #
-    #   set_dataset_property: [{
-    #     selector: "string",
-    #     name:     "string",
-    #     value:    "string"
-    #   }, ...],
-    # }
-    def initialize(name)
+    def initialize(name, tools)
       @name = name
+      @tools = tools
       @operations = stub
-    end
 
-    def clear
-      @operations = stub
+      tools.each do |tool|
+        define_singleton_method tool do |options = {}|
+          add_operation tool, options
+        end
+      end
     end
 
     def broadcast
       operations.select! { |_, list| list.present? }
       operations.deep_transform_keys! { |key| key.to_s.camelize(:lower) }
       ActionCable.server.broadcast name, "cableReady" => true, "operations" => operations
-      clear
-    end
-
-    def set_cookie(value)
-      add_operation(:set_cookie, {cookie: value})
-    end
```

Comment:
Serious question: did you read the last paragraph of the PR?

I don't mean that in a shitty way... I'm trying to understand if you missed what I said, or if what I said doesn't cover it and I'm missing something.

Quick recap: this is a minor breaking change but one I'm currently advocating for because it's not even documented with and the number of people currently using it could be 0 or 1 today and I'm not sure why it even had to be different in the first place. I **strongly** suspect that it was an oversight.

Reply:
> Quick recap: this is a minor breaking change but one I'm currently advocating for because it's not even documented with and the number of people currently using it could be 0 or 1 today and I'm not sure why it even had to be different in the first place. I **strongly** suspect that it was an oversight.

DOH, sorry, I missed that part of the text. I should probably read the whole thing before I start switching before text and code 😬